xrootd/authorization_test.go

```diff
@@ -727,37 +739,86 @@ func TestWriteCacheAuthFiles(t *testing.T) {
 }
 
 func TestWriteOriginScitokensConfig(t *testing.T) {
-	ctx, cancel, egrp := test_utils.TestContext(context.Background(), t)
-	defer func() { require.NoError(t, egrp.Wait()) }()
-	defer cancel()
+	tests := []struct {
+		Name         string
+		Capabilities []string
+		ExpectedCfg  string
+		InputCfg     string
+	}{
+		{
+			Name:         "OriginSciTokensOnlyReads",
+			Capabilities: []string{"Reads"},
+			ExpectedCfg:  string(scitokensOriginReadOutput),
+			InputCfg:     "",
+		},
+		{
+			Name:         "OriginSciTokensOnlyWrites",
+			Capabilities: []string{"Writes"},
+			ExpectedCfg:  string(scitokensOriginWriteOutput),
+			InputCfg:     "",
+		},
+		{
+			Name:         "OriginSciTokensReadsWrites",
+			Capabilities: []string{"Reads", "Writes"},
+			ExpectedCfg:  string(scitokensOriginReadWriteOutput),
+			InputCfg:     "",
+		},
+		{
+			Name:         "OriginScitokensMergeDiff",
+			Capabilities: []string{"Reads"},
+			ExpectedCfg:  string(monitoringOutput),
+			InputCfg:     string(toMergeOutput),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			ctx, cancel, egrp := test_utils.TestContext(context.Background(), t)
+			defer func() { require.NoError(t, egrp.Wait()) }()
+			defer cancel()
 
-	server_utils.ResetTestState()
-	dirname := t.TempDir()
-	os.Setenv("PELICAN_ORIGIN_RUNLOCATION", dirname)
-	defer os.Unsetenv("PELICAN_ORIGIN_RUNLOCATION")
-	config_dirname := t.TempDir()
-	viper.Set("Origin.SelfTest", true)
-	viper.Set("ConfigDir", config_dirname)
-	viper.Set("Origin.RunLocation", dirname)
-	viper.Set("Origin.Port", 8443)
-	viper.Set("Server.WebPort", 8444)
-	viper.Set("Server.Hostname", "origin.example.com")
-	viper.Set(param.Origin_StorageType.GetName(), string(server_structs.OriginStoragePosix))
+			server_utils.ResetTestState()
+			defer server_utils.ResetTestState()
 
-	err := config.InitServer(ctx, server_structs.OriginType)
-	require.NoError(t, err)
+			viper.Set("Origin.Port", 8443)
```

Comment:
I know a lot of this code is going to get hit with merge conflicts so I'm not sure if this comment will be relevant soon, but can you switch any of the `viper.Set`s you touch to use `param.XXX.GetName()` for the key (the way it's done in line 786 here)?